%*******************************************************
% Abstract
%*******************************************************
\pdfbookmark[3]{Abstract}{abstract}
\chapter{Abstract}

\small
The recent revolution in sequencing technologies has lead to a dramatic shift in biology, especially in the field of genetics and microbiology: complex and complicated analysis such as complete bacterial genome sequencing have become cost-effective and easily available; because of this shift, disciples like genomics, phenomics and comparative genomics were born. Microbiological studies are no longer limited to a single strain of a particular species, but can be applied to a vast range of strains, thus targeting the natural diversity inside a bacterial species, which can have a considerable impact on the many application of microbiology. One of these applications concerns agriculture, and specifically the plant-bacteria interaction, where bacterial species play a prominent role in agricultural sustainability and in the nitrogen cycle through the rhizobial symbiosis.

This thesis has been focused on the analysis of the genetic determinants of the natural phenotypic variability in the \textit{Sinorhizobium meliloti} symbiosis with \textit{Medicago} plants: in particular the differences in terms of plant-growth promotion and resistance to environmental stresses were addressed. The long term goal of such a search is to provide a reservoir of genetic elements for the improvement of the agricultural application of this bacterium, which has an environmental, social and economical impact on the future of agriculture.

The first part of the thesis is focused in the development of computational biology tools for the analysis of the high number of data obtained with the new technologies of the so-called \textit{omics} era: specifically a software for the improvement of bacterial draft genomes (CONTIGuator) and a software suite for the combined analysis of genomics and phenomics data (DuctApe) are presented, with their application to real world datasets.

The second part of the thesis is focused on the elicitation of the genetic determinants of plant-bacteria interaction inside the class \textit{Alphaproteobacteria}, where the largest part of the bacterial species interacting from plants that have been studied come from, trying to understand if there is a common genetic background for this interaction inside this bacterial class and to which extent it is distributed in other \textit{taxa}; indeed a series of genetics element were found to be common, especially for the symbiotic interaction.

The last part is focused on one of the alphaproteobacterial bacteria studied in the previous part (\textit{S. meliloti}); through comparative genomics approaches, the natural variability in the plant growth promotion phenotype is studied and a series of determinant factors are elicited, comprising patterns of presence/absence of key symbiotic genes, as well as regulatory features. Moreover, the functional evolution of the peculiar structure of the \textit{S. meliloti} genome of this species was analyzed to understand its origin and evolution, leading to a more general view on the evolution of rhizobial genomes and other large and complex bacterial genomes.

The overall thesis shows that is indeed feasible to use the huge amount of data made available by the \textit{omics} technologies to link genomics and phenomics variability; in particular, the biology of the plant-bacteria symbiosis and its impact on a sustainable agriculture: both the computational tools than the approach used in this thesis are indeed applicable to other species with complex phenotypes and a known natural variability, which is one of the key factors for a sustainable agriculture.